app/models/article.rb

```diff
@@ -213,13 +213,14 @@ class Article < ApplicationRecord
   def self.active_threads(tags = ["discuss"], time_ago = nil, number = 10)
```

Comment:
This method is becoming a bit unwieldy, but it also doesn't really need to be in the `Article` class, as self-contained. I know we don't use the query object pattern much, but we do have 2 in `app/queries` and I'd be very open to adding more of them. Something like:

```ruby
module Articles
  class ActiveThreadsQuery
    DEFAULT_OPTIONS = {
      tags: ["discuss"],
      number: 10
    }.with_indifferent_access.freeze

    def self.call(relation: Article.published, options: {})
      options = DEFAULT_OPTIONS.merge(options)
      ....
    end
  end
end
```

This would also allow us to move some of the magic numbers out into constants, e.g.

```ruby
MINIMUM_SCORE = -4
```

Then we'd have to update the call sites (only 2 from what I can tell) from ` Article.active_threads` to `Articles::ActiveThreadsQuery.call`.

Reply:
I was thinking the same thing and was going to write a RFC for refactoring this code later, but loved your suggestion and took the learning experience head-on! That said, I pulled `.active_threads` out of the `Article` model, removed the `.active_threads`-related code from the `articles_spec.rb`, and removed the `Article.active_threads` calls in both of the necessary views (`_homepage.html.erb` and `_sidebar.html.erb`). I then replaced all of that code with `Articles::ActiveThreadsQuery`, `active_threads_query_spec.rb`, and `Articles::ActiveThreadsQuery.call(...)` to follow the query object pattern per your suggestion. 😊  Let me know what you think, @citizen428!

Reply:
Great @juliannatetreault, this is exactly what I had in mind 🎉